.github/workflows/prepare_release.yml

```
@@ -0,0 +1,67 @@
+---
+
+name: Prepare Release
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: 'Module version to be released.'
+        required: true
+        type: string
+      working-directory:
+        description: The working directory where all jobs should be executed. Used for modules in subdirectories like a monorepo or a control repository.
+        default: '.'
+        required: false
+        type: string
+      base-branch:
+        description: 'The branch that will be used as the origin for the release branch.'
+        required: false
+        default: 'master'
+        type: string
+
+env:
+  BUNDLE_WITHOUT: development:test:system_tests
+  BUNDLE_WITH: release
+  GIT_AUTHOR_NAME: Release Automation
+  GIT_AUTHOR_EMAIL: "${{ github.repository_owner }}@users.noreply.github.com"
+  BLACKSMITH_FULL_VERSION: "${{ inputs.version }}"
+
+jobs:
+  prepare_release:
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    name: 'Puppet Forge'
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.base-branch }}
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.3'
+          bundler-cache: true
+          working-directory: ${{ inputs.working-directory }}
+      - name: Update metadata.json to new version
+        run: bundle exec rake module:bump:full
```

One way to implement https://github.com/voxpupuli/gha-puppet/pull/66/files#r1892651518 is to bump the patch level if it isn't given. While it may not be perfect, it means you don't need to specify it which is commonly what you do on the first run.
```suggestion
        run: |
          if [[ -n BLACKSMITH_FULL_VERSION ]] ; then
            bundle exec rake module:bump:full
          else
            bundle exec rake module:bump
          fi
```